components/notebook-controller/go.mod

```diff
@@ -1,8 +1,7 @@
 module github.com/kubeflow/kubeflow/components/notebook-controller
 
 go 1.21
-
-toolchain go1.21.9
+toolchain go1.23.7
```

Comment:
I have no idea what this will do. Looking forward to see the CI on it.

Reply:
```
[1/2] STEP 11/11: RUN if [ -z ${CACHITO_ENV_FILE} ]; then go mod download; else source ${CACHITO_ENV_FILE}; fi &&   CGO_ENABLED=1 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -tags strictfipsruntime -a -o ./bin/manager main.go
go: golang.org/x/net@v0.36.0: module golang.org/x/net@v0.36.0 requires go >= 1.23.0 (running go 1.21.13; GOTOOLCHAIN=local)
```

so sad

Reply:
With that said, we should initiate golang upgrade to whole pipeline I guess?

Reply:
We can't, we don't have go-builder image for 1.23, latest is 1.22

Reply:
Opened Slack thread, hopefully somebody will have some comments. Until then, probably nothing to do here

https://redhat-internal.slack.com/archives/C05NXTEHLGY/p1741854502977309

Reply:
Conclusion is that we should be able to ignore this for now, until we can upgrade to go 1.23 easily. Should not be 2.19 release blocker.

Reply:
can we , we try go 1.22 , and see if the vulnerability can be fixed in this golang version ?
is there a reason to upgrade to 1.23 ?

Reply:
we can try go 1.22, but it won't help with x/net

go 1.22 is a good thing to try anyways, when there is a bit of slack time